src/DL/TransactionHistoryHandler.java

```diff
@@ -0,0 +1,77 @@
+package DL;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class TransactionHistoryHandler {
+
+    private final String path;
+
+
+    public TransactionHistoryHandler(String path) {
+        create(path);
+        this.path = path;
+    }
+
+
+    private boolean create(String path) {
+        File file = new File(path);
+        try {
+            return file.createNewFile();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+            return false;
+        }
+    }
+
+    public ArrayList<Transaction> read(){
+        ArrayList<Transaction> result = new ArrayList<>();
+        try(Scanner scanner = new Scanner(new File(this.path))){
+            while (scanner.hasNextLine()) {
+                String newTask = scanner.nextLine();
+                String[] propertiesOfTransaction = newTask.split(";");
```

Comment:
it's not a critical issue, but _properties_ is also a good option for the name of this variable. but it is much shorter and still understandable

Reply:
fixed